### PR TITLE
Add clustermapping API as a source on Udaman (UA-584)

### PR DESCRIPTION
**Pull request by vcward**
Added functionality to retrieve data from [Cluster Mapping API](http://clustermapping.us/). There is documentation added to the help page with an example of what the eval statement should look like when adding it as a source to a series. Users can specify what data they want (i.e. wage, employment, location quotient, etc.) for a given region (i.e. State of Hawaii).

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,6 +34,7 @@ gem 'rails-assets-select2', '~> 4.0.5', source: 'https://rails-assets.org'
 gem 'ancestry'
 gem 'will_paginate', '~> 3.1.1'
 gem 'font-awesome-sass', '~> 5.6', '>= 5.6.1'
+gem 'rb-fsevent'
```
Comment (vcward) [inline Gemfile]: Added this to get the rails server running

File: app/controllers/data_sources_controller.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def toggle_reload_nightly
   end
 
   def new
-    @series = Series.find data_source_params[:series_id]
+    @series = Series.find params[:series_id]
```
Comment (vcward) [inline app/controllers/data_sources_controller.rb]: And this change is to fix https://uhero-analytics.atlassian.net/browse/UA-1090

Comment (kotodharma) [reply app/controllers/data_sources_controller.rb]: Thanks for catching this bug!

**Comment by kotodharma** [comment]
One more note, now that I've actually looked at the clustermapping API output: Are we only ever going to be retrieving numeric values from this? If not, we might want to have the code handle the different value types appropriately.
Just to be clear, I didn't approve just because I thought we should discuss the few things I've brought up first, but as soon as we clear that stuff, we can go ahead.

**Comment by vcward** [comment]
Thanks!
Re: the Gemfile, yeah I had ran `gem install rb-fsevent`, and it was still giving me trouble. I'll play around with it some more though.
The db schema thing was not on purpose. I ran a migration at some point, but I'm not 100% sure why that change happened. And you're right about the frequency; the clustermapping series are annual observations.
I think we would only be using the numeric values from the API (should probably confirm this with Carl though). We did a report a while back using this data and pretty much focused on their values for 'location quotient' and employment.

**Pull request merged**